actions/contacts/search.go

```diff
@@ -72,3 +72,116 @@ func (a *Action) search(c *gin.Context) {
+// @Tags		Contacts
+// @Produce		json
+// @Param		SearchContacts body filter.SearchContacts false "Supports targeted resource searches with filters and metadata, plus options for pagination and sorting to streamline data exploration and analysis"
+// @Success		200 {object} models.PageModel[models.Contact] "Page of contacts"
+// @Failure		400	"Bad request - Error while parsing SearchContacts from request body"
+// @Failure 	500	"Internal server error - Error while searching for contacts"
+// @Router		/v1/contacts/ [GET]
+// @Security	x-auth-xpub
+func (a *Action) getContacts(c *gin.Context) {
+	reqXPubID := c.GetString(auth.ParamXPubHashKey)
+
+	var reqParams filter.SearchContacts
+	if err := c.Bind(&reqParams); err != nil {
+		c.JSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	conditions, err := reqParams.Conditions.ToDbConditions()
+	if err != nil {
+		c.JSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	reqParams.DefaultsIfNil()
+
+	contacts, err := a.Services.SpvWalletEngine.GetContactsByXpubID(
+		c.Request.Context(),
+		reqXPubID,
+		mappings.MapToMetadata(reqParams.Metadata),
+		conditions,
+		mappings.MapToQueryParams(reqParams.QueryParams),
+	)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	contracts := mappings.MapToContactContracts(contacts)
+
+	count, err := a.Services.SpvWalletEngine.GetContactsByXPubIDCount(
+		c.Request.Context(),
+		reqXPubID,
+		mappings.MapToMetadata(reqParams.Metadata),
+		conditions,
+	)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	response := models.PageModel[models.Contact]{
+		Content: contracts,
+		Page: models.PageDescription{
+			Size:          len(contracts),
+			Number:        0,
+			TotalElements: int(count),
+			TotalPages:    len(contracts) / int(count),
+		},
+	}
+
+	c.JSON(http.StatusOK, response)
+}
+
+// getContactsByID will fetch a list of contacts
+// Get contacts godoc
+// @Summary		Get contacts by id
+// @Description	Get contacts by id
+// @Tags		Contacts
+// @Produce		json
+// @Success		200 {object} models.Contact "Contact"
+// @Failure		400	"Bad request - Error while parsing SearchContacts from request body"
+// @Failure 	500	"Internal server error - Error while searching for contacts"
+// @Router		/v1/contacts/{id} [GET]
+// @Security	x-auth-xpub
+func (a *Action) getContactsByID(c *gin.Context) {
+	reqXPubID := c.GetString(auth.ParamXPubHashKey)
+	paymail := c.Param("paymail")
+	var reqParams filter.SearchContacts
+	reqParams.ConditionsModel.Conditions.Paymail = &paymail
+
+	conditions, err := reqParams.Conditions.ToDbConditions()
+	if err != nil {
+		c.JSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	reqParams.DefaultsIfNil()
+
+	contacts, err := a.Services.SpvWalletEngine.GetContactsByXpubID(
+		c.Request.Context(),
+		reqXPubID,
+		mappings.MapToMetadata(reqParams.Metadata),
+		conditions,
+		mappings.MapToQueryParams(reqParams.QueryParams),
+	)
```

Comment:
By `ID` or by `paymail` 🤔 ?

Reply:
Thanks. Fixed. It should be by `paymail`